transforms/universal/fdedup/python/src/data_cleaning_transform.py

```diff
@@ -0,0 +1,179 @@
+""" Name of the column storing the unique ID assigned to each document"""
+duplicate_list_location_cli_param = f"{cli_prefix}{duplicate_list_location_key}"
+""" Location of the list of duplicate documents marked for removal"""
+operation_mode_cli_param = f"{cli_prefix}{operation_mode_key}"
+""" Operation mode, can be one of 'filter_duplicates', 'filter_non_duplicates', or 'annotate'"""
+
+captured_arg_keys = [
+    document_id_column_key,
+    duplicate_list_location_key,
+    operation_mode_key,
+]
+
+# defaults
+document_id_column_default = "int_id_column"
+""" Default name of the column storing the unique ID assigned to each document"""
+duplicate_list_location_default = os.path.join("docs_to_remove_consolidated", "docs_to_remove_consolidated.parquet")
+""" Default location of the list of duplicate documents marked for removal"""
+operation_mode_default = "filter_duplicates"
+""" Default value for operation mode, will filter out all the duplicate documents"""
+
+dataclean_data_factory_key = "dc_data_factory"
+dataclean_data_access_key = "dc_data_access"
+
+
+class DataCleaningTransform(AbstractTableTransform):
+    """
+    This is the third transform of the fuzzy dedup pipeline. It takes as input
+    the list of the documents to remove (identified as duplicates during the
+    cluster analysis phase, and the original dataset. Each dataset file is
+    imported into a table, and the documents that are in the documents to remove
+    list are filtered out from that table. The output is a new dataset, which
+    keeps the directory structure of the input dataset, but has all the fuzzy
+    duplicates removed.
+
+    Args:
+        duplicate_location: location (local or s3) of the duplicate document list
+    """
+
+    def __init__(self, config: dict[str, Any]):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments
+        defined by the companion runtime, ClusterAnalysisTransformRuntime.
+        """
+        super().__init__(config)
+        self.logger = get_logger(__name__)
+        self.document_id_column = config.get(document_id_column_key, document_id_column_default)
+        self.duplicate_list_location = config.get(duplicate_list_location_key, duplicate_list_location_default)
+        self.operation_mode = config.get(operation_mode_key, operation_mode_default)
+        contents = config.get("df")
+        self.docs_to_remove_df = pl.read_parquet(io.BytesIO(contents))
+        self.logger.info(f"Got docs_to_remove_df with {len(self.docs_to_remove_df)} rows")
+        self.docs_to_remove_df = self.docs_to_remove_df.rename({"docs_to_remove": self.document_id_column})
+
+    def transform(self, table: pa.Table, file_name: str = None) -> tuple[list[pa.Table], dict[str, Any]]:
+        self.logger.info(f"Transforming table with {table.num_rows} rows from file {file_name}")
```

Comment:
maybe make this debug so the kfp log doesn't get overwhelmed.

Reply:
This is the only log message I output in this transform, and it is invoked once per file.

Reply:
Fixed in commit https://github.com/IBM/data-prep-kit/pull/699/commits/8de14b51e55a69bfa7f4dddc763500fb3a01db4d